backend/src/services/pattern_extraction_service.py

```
@@ -1,134 +1,698 @@
+    def extract_patterns_from_description(self, description: str) -> List[PatternSuggestion]:
+        """Alias for generate_patterns_from_description for backward compatibility"""
+        return self.generate_patterns_from_description(description)
     
     def create_category_with_rule(
         self, 
         category_name: str, 
         category_type: str, 
         pattern: str, 
         field_to_match: str = 'description',
-        condition: MatchCondition = MatchCondition.CONTAINS
+        condition = None  # Will be MatchCondition from models.category
```

Flake8 (7.2.0)

[error] unexpected spaces around keyword / parameter equals

(E251)